apm-sniffer/bytebuddy-patch/src/main/java/org/apache/skywalking/apm/agent/bytebuddy/SWClassFileLocator.java

```diff
@@ -0,0 +1,178 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.apm.agent.bytebuddy;
+
+import net.bytebuddy.dynamic.ClassFileLocator;
+import org.apache.skywalking.apm.agent.core.logging.api.ILog;
+import org.apache.skywalking.apm.agent.core.logging.api.LogManager;
+
+import java.io.IOException;
+import java.lang.instrument.Instrumentation;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.LinkedBlockingDeque;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Resolve auxiliary type
+ */
+public class SWClassFileLocator implements ClassFileLocator {
```

Comment:
Is this class copied and changed from byte-buddy upstream? If so, we should declare it.
Because there is a lot of class resolution logic, we seem never used. It looks like bytebuddy default behavior.

Reply:
`SWClassFileLocator`  is used to resovle auxiliary type cross `AgentBuilders`, it is unused currently.

Reply:
Could you be more specific about the workflow? Such as why new thread, and when do we need this?

Reply:
Get class bytecode from separate thread to bypass jdk limitation or bug:  https://github.com/raphw/byte-buddy/issues/1434.  Since we haven't used it yet, I'll move it to the test code for now.